packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts

```diff
@@ -1,28 +1,125 @@
 import { memoizeFunction } from '../../Utilities';
-import { mergeStyleSets, ITheme } from '../../Styling';
-import { IContextualMenuStyles } from './ContextualMenu.Props';
-
+import { ITheme, mergeStyleSets } from '../../Styling';
+import { IContextualMenuStyles, IMenuItemStyles } from './ContextualMenu.Props';
+import { getStyles as getContextualMenuStyles, getMenuItemStyles } from './ContextualMenu.styles';
 export interface IContextualMenuClassNames {
-  title?: string;
+  container: string;
+  root: string;
+  list: string;
+  divider: string;
+  header: string;
+  title: string;
+}
+
+export interface IMenuItemClassNames {
+  item: string;
+  root: string;
+  linkContent: string;
+  icon: string;
+  subMenuIcon: string;
+  label: string;
 }
 
-export const getClassNames = memoizeFunction((
-  styles: IContextualMenuStyles,
+export const getContextualMenuClassNames = memoizeFunction((
   theme: ITheme,
+  className: string
 ): IContextualMenuClassNames => {
 
-  styles = styles || {};
+  const styles = getContextualMenuStyles(theme);
 
   return mergeStyleSets({
-    title: [{
-      fontSize: '16px',
-      paddingRight: '14px',
-      paddingLeft: '14px',
-      paddingBottom: '5px',
-      paddingTop: '5px',
-      backgroundColor: theme.palette.neutralLight
-    },
-    styles.title
-    ]
+    container: [
+      'ms-ContextualMenu-container',
+      className,
+      styles.container
+    ],
+    root: [
+      'ms-ContextualMenu is-open',
+      styles.root
+    ],
+    list: [
+      'ms-ContextualMenu-list is-open',
+      styles.list
+    ],
+    divider: [
+      'ms-ContextualMenu-divider',
+      className,
```

Comment:
It's kind of strange that the divider and container both get the ame className

Reply:
Yeah.. i just saw this bug. The separator's classname comes from item.classname so I am moving to the getItemClassNames